File: my-folder/0202-happy-number/solution.cpp
class Solution {
public:

    int solve(int n){
        int sum=0;
       
            while(n!=0){
                int num = n%10;
                sum += num*num;
                n=n/10;
        }
        return sum;
    }

    bool isHappy(int n) {
        unordered_set<int> set;
        while(n!=1 && !set.count(n)){
            set.insert(n);
            n = solve(n);
        }
        return n==1;
    }
};

File: my-folder/0202-happy-number/solution.py
class Solution:
   
    def isHappy(self, n: int) -> bool:
        return self.solve(n, [])

    def solve(self, n: int, track: List[int]) -> bool:
        if n==1:
            return True
        res=sum(int(i)**2 for i in str(n))
        if res in track:
            return False
        else:
            track.append(res)
        return self.solve(res, track)
    
        
        
            

        

